specs/interop/token-bridging.md

```diff
@@ -2,96 +2,298 @@
+  (bool success, ) = _to.call(_data);
+  require(success, "External call failed");
+}
 
-    _mint(to, amount);
-  }
+/////// Internal functions ///////////
+function _sendMessage(uint256 _destination, bytes memory _data) internal virtual {
+  L2ToL2CrossDomainMessenger.sendMessage(_destination, address(this), _data);
+}
+```
+
+Note: Some other naming options that were considered were `bridgeERC20`, `send()`, `xTransfer()`, `Transfer` (with a chainId parameter).
+
+## Invariants
+
+Besides the ERC20 invariants, the SuperchainERC20 will require the following interop specific properties:
+
+- Conservation of finalized `totalSupply`: The minted `amount` in `finalizeSendERC20()` should match the `amount` that was burnt in `sendERC20()`, as long as target chain has the initiating chain in the dependency set.
+  - Corollary 1: Cross-chain transactions will conserve the sum of `totalSupply` for each chain in the Superchain across safe blocks.
+  - Corollary 2: Each initiated but not finalized message (included in initiating chain but not yet in target chain) will decrease the `totalSupply` exactly by the burnt `amount`.
+  - Corollary 2: `SuperchainERC20s` should not charge a token fee or increase the balance when moving cross-chain.
+  - Question: what should we assume for chains not in the dependency set? a rollback method could modify this invariant to also consider that case.
+- Freedom of movement: Users should be able to send tokens into any target chain that has initiating chain in its dependency set.
+  - Question: should we add "deployed" to the invariant or assume its not going to be an issue?
+- [To discuss] Unique Messenger: The `sendERC20()` functions must exclusively use the `L2toL2CrossDomainMessenger` for messaging. Similarly, the `finalizeSendERC20()` function should only process messages originating from the L2toL2CrossDomainMessenger.
+  - Corollary: xERC20 and other standards from third party bridges should use different functions.
+- [To discuss] Locally initiated: The bridging action should be initialized from the chain where funds are located only.
+  - This is because same address might correspond to different users cross-chain. For example, two SAFEs with the same address in two chains might have different owners. It it possible to distiguish an EOA from a contract by checking the size of the code in the caller's address, but this will probably change with [EIP-7702](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md).
+  - A way to allow for remotely initiated bridging is to include remote approval, i.e. approve a certain address in a certain chainId to spend local funds.
+- [To discuss] Bridge Event: `sendERC20()` should emit a `Bridged` event.
+  - We already have a lot of events triggering from these actions. Should we include yet another one?
+
+**Desired properties**
+
+- Spatial symmetry: Same implementation and address across chains.
+
+**Open questions**
+
+- Should we define invariants for upgrades?
```

Comment:
We can define the upgrade in the future, i think its fine to remove open q's. This defines the interface and the convert approach is one usage of the interface that we will implement first